core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -540,6 +542,13 @@ class Log(@volatile var logStartOffset: Long,
     }
   }, period = producerIdExpirationCheckIntervalMs, delay = producerIdExpirationCheckIntervalMs, unit = TimeUnit.MILLISECONDS)
 
+  // For compatibility, metrics are defined to be under `Log` class
+  override def metricName(name: String, tags: scala.collection.Map[String, String]): MetricName = {
+    val klass = getClass
+    val pkg = if (klass.getPackage == null) "" else klass.getPackage.getName
```

Comment:
nit: it seems better to do the following? maybe a different variable name after the change.
```suggestion
    val klass = getClass.getPackage
    val pkg = if (klass == null) "" else klass.getName
```

Reply:
Addressed in 5a85097